src/cdk-experimental/testing/testbed.ts

```diff
@@ -0,0 +1,217 @@
+  const root = fixture.nativeElement;
+  const stabilize = async () => {
+    fixture.detectChanges();
+    await fixture.whenStable();
+  };
+  const locator = new UnitTestLocator(root, stabilize);
+  return new componentHarness(locator);
+}
+
+/**
+ * Gets the corresponding Element for the root of a TestElement.
+ */
+export function getNativeElement(testElement: TestElement): Element {
+  if (testElement instanceof UnitTestElement) {
+    return testElement.element;
+  }
+
+  throw new Error('Invalid element provided');
+}
+
+/**
+ * Locator implementation for testbed.
+ * Note that, this locator is exposed for internal usage, please do not use it.
+ */
+export class UnitTestLocator implements Locator {
+  private _rootElement: TestElement;
+
+  constructor(private _root: Element, private _stabilize: () => Promise<void>) {
+    this._rootElement = new UnitTestElement(_root, this._stabilize);
+  }
+
+  host(): TestElement {
+    return this._rootElement;
+  }
+
+  async find(css: string, options?: Options): Promise<TestElement|null> {
+    await this._stabilize();
+    const e = getElement(css, this._root, options);
+    if (e === null) {
+      return null;
+    }
+    return new UnitTestElement(e, this._stabilize);
+  }
+
+  async findAll(css: string): Promise<TestElement[]> {
+    await this._stabilize();
+    const elements = this._root.querySelectorAll(css);
+    const res: TestElement[] = [];
+    for (let i = 0; i < elements.length; i++) {
+      res.push(new UnitTestElement(elements[i], this._stabilize));
+    }
+    return res;
+  }
+
+  async load<T extends ComponentHarness>(
+    componentHarness: ComponentHarnessType<T>, css: string,
+    options?: Options): Promise<T|null> {
+    const root = getElement(css, this._root, options);
+    if (root === null) {
+      return null;
+    }
+    const locator = new UnitTestLocator(root, this._stabilize);
+    return new componentHarness(locator);
+  }
+
+  async loadAll<T extends ComponentHarness>(
+    componentHarness: ComponentHarnessType<T>,
+    rootSelector: string): Promise<T[]> {
+    await this._stabilize();
+    const roots = this._root.querySelectorAll(rootSelector);
+    const res: T[] = [];
+    for (let i = 0; i < roots.length; i++) {
+      const root = roots[i];
+      const locator = new UnitTestLocator(root, this._stabilize);
+      res.push(new componentHarness(locator));
+    }
+    return res;
+  }
+}
+
+class UnitTestElement implements TestElement {
+  constructor(
+    readonly element: Element, private _stabilize: () => Promise<void>) {}
+
+  async blur(): Promise<void> {
+    await this._stabilize();
+    (this.element as HTMLElement).blur();
+    await this._stabilize();
+  }
+
+  async clear(): Promise<void> {
+    await this._stabilize();
+    if (!(this.element instanceof HTMLInputElement ||
+      this.element instanceof HTMLTextAreaElement)) {
+      throw new Error('Attempting to clear an invalid element');
+    }
+    this.element.focus();
```

Comment:
Moving focus here might a bit risky since it's not what would necessarily happen to a real user. E.g. an input "clear" button won't move focus into the input.

Reply:
Is that true? in Chrome with an `<input type="search">` pressing the button also focuses.

Reply:
I meant if it's a custom clear button, rather than the native browser one.

Reply:
Ah, in that case the particular component that implements the clear button can choose to expose a `clear` method in its harness API that clicks the custom button instead. This class is meant to be an implementation of the interface harness authors use to write their harnesses.